File: frontend/src/types/recommendation.ts
import type { Priority } from "./common"

export type RecommendationStatus = "Pending" | "Accepted" | "Rejected"

export interface Recommendation {
  id: string
  customerId: string
  customerName: string
  product: string
  priority: Priority
  reason: string
  triggeredAgent: string
  segment: string
  country: string
  status: RecommendationStatus
  generatedAt: string
}

export interface RecommendationListResponse {
  items: Recommendation[]
  total: number
}

export interface RecommendationListParams {
  status?: string
  priority?: string
  segment?: string
  country?: string
  triggeredAgent?: string
  page?: number
  pageSize?: number
}

File: backend/app/schemas/recommendation.py
from typing import Literal
from pydantic import BaseModel

RecommendationStatus = Literal["Pending", "Accepted", "Rejected"]


class Recommendation(BaseModel):
    id: str
    customerId: str
    customerName: str
    product: str
    priority: Literal["High", "Medium", "Low"]
    reason: str
    triggeredAgent: str
    segment: str
    country: str
    status: RecommendationStatus
    generatedAt: str


class RecommendationListResponse(BaseModel):
    items: list[Recommendation]
    total: int
